### PR TITLE
Added getMempoolInfo RPC

### PR DESCRIPTION
**Pull request by aikchun**
#### What kind of PR is this?:

/kind feature

#### What this PR does / why we need it:
Implemented RPC for #48 
1. blockchain.getmempoolinfo

### CODE REVIEW

**Review by thedoublejay — DISMISSED** [review]

File: packages/jellyfish-api-core/src/category/blockchain.ts [hunk 1]
```diff
@@ -367,3 +376,13 @@ export interface BlockStats {
...
 }
+
+export interface MempoolInfo {
+  loaded: boolean
+  size: number
+  bytes: number
+  usage: number
+  maxmempool: number
+  mempoolminfee: number
+  minrelaytxfee: number
```
Comment (canonbrother) [inline packages/jellyfish-api-core/src/category/blockchain.ts:387]: size, bytes, usage, maxmempool are int64 -> use bignumber
fee is amount -> use bignumber as well

**Review by canonbrother — CHANGES_REQUESTED** [review]

File: packages/jellyfish-api-core/src/category/blockchain.ts [hunk 2]
```diff
@@ -181,6 +181,15 @@ export class Blockchain {
...
     return await this.client.call('getbestblockhash', [], 'number')
   }
+
+  /**
+   * Returns details on the active state of the TX memory pool.
+   *
+   * @return {Promise<MempoolInfo>}
+   */
+  async getMempoolInfo (): Promise<MempoolInfo> {
+    return await this.client.call('getmempoolinfo', [], 'number')
```
Comment (jingyi2811) [inline packages/jellyfish-api-core/src/category/blockchain.ts:191]: You need to change `number` to `bignumber` in order to get the result in bignumber

File: website/docs/jellyfish/api/blockchain.md [hunk 3]
```diff
@@ -268,6 +269,26 @@ interface MempoolTx {
 }
 ```
 
+## getMempoolInfo
+
+Returns details on the active state of the TX memory pool.
+
+```ts="client.blockchain.getMempoolInfo"
+interface blockchain {
+  getMempoolInfo(): Promise<MempoolInfo>
```
Comment (jingyi2811) [inline website/docs/jellyfish/api/blockchain.md:278]: Add a space between `getMempoolInfo` and `()`

File: packages/jellyfish-api-core/src/category/blockchain.ts (hunk 1 above)
Comment (aikchun) [reply packages/jellyfish-api-core/src/category/blockchain.ts:387]: Change the precision to `bignumber`.

File: website/docs/jellyfish/api/blockchain.md (hunk 3 above)
Comment (aikchun) [reply website/docs/jellyfish/api/blockchain.md]: Added space in between the method name and the parenthesis.

**Review by jingyi2811 — DISMISSED** [review]

File: packages/jellyfish-api-core/__tests__/category/blockchain/getMempoolInfo.test.ts [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+  it('should getMempoolInfo', async () => {
+    const mempoolInfo = await client.blockchain.getMempoolInfo()
+
+    expect(typeof mempoolInfo.loaded).toStrictEqual('boolean')
+    expect(mempoolInfo.size).toBeInstanceOf(BigNumber)
+    expect(mempoolInfo.bytes).toBeInstanceOf(BigNumber)
+    expect(mempoolInfo.usage).toBeInstanceOf(BigNumber)
+    expect(mempoolInfo.maxmempool).toBeInstanceOf(BigNumber)
+    expect(mempoolInfo.mempoolminfee).toBeInstanceOf(BigNumber)
+    expect(mempoolInfo.minrelaytxfee).toBeInstanceOf(BigNumber)
```
Comment (canonbrother) [inline packages/jellyfish-api-core/__tests__/category/blockchain/getMempoolInfo.test.ts:28]: this `assert` alerted me
can you declare `mempoolInfo` with type??
```
const mempoolInfo: MempoolInfo = ...
```

**Review by canonbrother — CHANGES_REQUESTED** [review]

File: packages/jellyfish-api-core/__tests__/category/blockchain/getMempoolInfo.test.ts (hunk 4 above)
Comment (aikchun) [reply packages/jellyfish-api-core/__tests__/category/blockchain/getMempoolInfo.test.ts:29]: Added the type annotation.

Comment (canonbrother) [reply packages/jellyfish-api-core/__tests__/category/blockchain/getMempoolInfo.test.ts:29]: I actually expect you will get error after specific the type but nope
TBH, this assertion is not **type safe**
`expect(mempoolInfo.size).toBeInstanceOf(BigNumber)`
please change to 
`expect(mempoolInfo.size instanceof BigNumber).toStrictEqual(true)`

**Review by canonbrother — CHANGES_REQUESTED** [review]

File: packages/jellyfish-api-core/__tests__/category/blockchain/getMempoolInfo.test.ts (hunk 4 above)
Comment (aikchun) [reply packages/jellyfish-api-core/__tests__/category/blockchain/getMempoolInfo.test.ts:29]: `instanceof` expects the left hand value to be of `any` or object type.

```
expect(mempoolInfo.size instanceof BigNumber).toStrictEqual(true)

    packages/jellyfish-api-core/__tests__/category/blockchain/getMempoolInfo.test.ts:25:12 - error TS2358: The left-hand side of an 'instanceof' expression must be of type 'any', an object type or a type parameter.
```

We could do this only if we don't type annotate `const mempoolInfo`

```
// We can't do instanceof if we do this
const mempool: MempoolInfo = await client.blockchain.getMempoolInfo()

expect(mempool.size instanceof BigNumber).toStrictEqual(true)
```

```
// we can do this
const mempool = await client.blockchain.getMempoolInfo()

expect(mempool.size instanceof BigNumber).toStrictEqual(true)
```

File: packages/jellyfish-api-core/src/category/blockchain.ts (hunk 1 above)
Comment (canonbrother) [inline packages/jellyfish-api-core/src/category/blockchain.ts:387]: you gonna change all these to bignumber

File: packages/jellyfish-api-core/__tests__/category/blockchain/getMempoolInfo.test.ts (hunk 4 above)
Comment (canonbrother) [reply packages/jellyfish-api-core/__tests__/category/blockchain/getMempoolInfo.test.ts:29]: that's the safety way to test
you gonna update the interface
https://github.com/DeFiCh/jellyfish/pull/364/files#r648974024

Comment (fuxingloh) [reply packages/jellyfish-api-core/__tests__/category/blockchain/getMempoolInfo.test.ts:29]: Some context for reference 

https://github.com/DeFiCh/jellyfish/issues/18
https://github.com/DeFiCh/jellyfish/pull/47
https://github.com/DeFiCh/jellyfish/issues/93
https://github.com/DeFiCh/jellyfish/issues/104
https://github.com/DeFiCh/jellyfish/pull/128

Comment (aikchun) [reply packages/jellyfish-api-core/__tests__/category/blockchain/getMempoolInfo.test.ts]: Made the necessary changes and updated the docs.

**Review by canonbrother — APPROVED** [review]

**Review by jingyi2811 — APPROVED** [review]

**Pull request merged**